File: TrackBudget/TrackBudget.Domain/Entities/Transfer.cs
using TrackBudget.Domain.Common;

namespace TrackBudget.Domain.Entities;

public class Transfer : BaseEntity
{
    public decimal Amount { get; set; }

    public DateTime Date { get; set; }

    public string? Notes { get; set; }

    public Guid UserId { get; set; }

    public User User { get; set; } = null!;

    public Guid FromAccountId { get; set; }

    public Account FromAccount { get; set; } = null!;

    public Guid ToAccountId { get; set; }

    public Account ToAccount { get; set; } = null!;
}

File: TrackBudget/TrackBudget.Client/src/types/transfer.ts
export interface Transfer {
  accountId: string;
  amount: number;
  date: string;
  notes: string | null;
  fromAccountId: string;
  fromAccountName: string;
  toAccountId: string;
  toAccountName: string;
  createdAt: string;
}

export interface CreateTransfer {
  amount: number;
  date: string;
  notes: string | null | undefined;
  fromAccountId: string;
  toAccountId: string;
}
